src/main/java/io/pravega/connectors/flink/Pravega.java

```diff
@@ -0,0 +1,332 @@
+            extends AbstractStreamingReaderBuilder<Row, TableSourceReaderBuilder> {
+
+        private DeserializationSchema<Row> deserializationSchema;
+
+        @Override
+        protected DeserializationSchema<Row> getDeserializationSchema() {
+            return this.deserializationSchema;
+        }
+
+        @Override
+        protected TableSourceReaderBuilder builder() {
+            return this;
+        }
+
+        /**
+         * Pass the deserialization schema to be used.
+         * @param deserializationSchema the deserialization schema.
+         * @return TableSourceReaderBuilder instance.
+         */
+        protected TableSourceReaderBuilder withDeserializationSchema(DeserializationSchema<Row> deserializationSchema) {
+            this.deserializationSchema = deserializationSchema;
+            return this;
+        }
+
+        /**
+         * factory to build an {@link FlinkPravegaInputFormat} for using Table API in a batch environment.
+         * @return a supplier to eagerly validate the configuration and lazily construct the input format.
+         */
+        FlinkPravegaInputFormat<Row> buildInputFormat() {
+            Preconditions.checkState(deserializationSchema != null, "The deserializationSchema must be provided.");
+            final List<StreamWithBoundaries> streams = resolveStreams();
+            final ClientConfig clientConfig = getPravegaConfig().getClientConfig();
+            return new FlinkPravegaInputFormat<>(clientConfig, streams, deserializationSchema);
+        }
+    }
+
+    /**
+     * Writer builder which can be used to define the Pravega writer configurations. The supplied configurations will be used
+     * to create appropriate Table sink implementation.
+     *
+     */
+    public static class TableSinkWriterBuilder<T extends AbstractStreamingWriterBuilder>
+            extends AbstractStreamingWriterBuilder<Row, TableSinkWriterBuilder> {
+
+        private String routingKeyFieldName;
+
+        private SerializationSchema<Row> serializationSchema;
+
+        /**
+         * Sets the field name to use as a Pravega event routing key.
+         * @param fieldName the field name.
+         */
+        public TableSinkWriterBuilder withRoutingKeyField(String fieldName) {
+            this.routingKeyFieldName = fieldName;
+            return builder();
+        }
+
+        /**
+         * Pass the serialization schema to be used.
+         * @param serializationSchema the serialization schema.
+         */
+        public TableSinkWriterBuilder withSerializationSchema(SerializationSchema<Row> serializationSchema) {
+            this.serializationSchema = serializationSchema;
+            return builder();
+        }
+
+        @Override
+        protected TableSinkWriterBuilder builder() {
+            return this;
+        }
+
+        /**
+         * Creates the sink function based on the given table sink configuration and current builder state.
+         *
```

Comment:
Minor: let's be consistent in the spacing between the method description and the params in Javadoc comments.

Reply:
fixed